wandb/sdk/launch/create_job.py

```diff
@@ -168,6 +168,7 @@ def _create_job(
         return None, "", []
 
     job_builder = _configure_job_builder_for_partial(tempdir.name, job_source=job_type)
+    job_builder._settings.update(job_name=name)
```

Comment:
also necessary to maintain behavior of `_create_job()`.  See note about overloading `name` in https://github.com/wandb/wandb/pull/8575/files#r1797254733